### DISCUSSION TITLE
How do we implement SCA with this module? createVerificationDetails doesn't seem to return a verification token

### DISCUSSION THREAD

**Discussion by byang14** [weareseeed/react-square-web-payments-sdk#42] [Help] [answered]
How does SCA work with this module for credit card payments?

I notice there is a createVerificationDetails where we need to pass in info like this:
createVerificationDetails={() => ({
      amount: '1.00',
      /* collected from the buyer */
      billingContact: {
        addressLines: ['123 Main Street', 'Apartment 1'],
        familyName: 'Doe',
        givenName: 'John',
        countryCode: 'GB',
        city: 'London',
      },
      currencyCode: 'GBP',
      intent: 'CHARGE',
    })}

However, it doesn't seem to return a verificationToken for BE to use. 

Is that automatically done? I've tested it and it does not seem like a verificationToken is accessible anywhere.

**Comment by danestves** [comment, marked as answer]
Hi @byang14 can you make a reproduction of this problem on https://react.new?

**Reply by byang14** [reply]
Where is the verification token saved after createVerificationDetails is called? And how do we access the verification token when we called cardTokenizeResponseReceived?

I realize we get token, buyer in cardTokenizeResponseReceived but token.token is the sourceId, not the verification token

**Reply by danestves** [reply]
```tsx
    cardTokenizeResponseReceived={(token, buyer) => {
      console.info({ token, buyer })
    }}
```

**Reply by byang14** [reply]
Looks like the verification token is in the buyer. Thanks

**Reply by danestves** [reply]
Yeah, because is the buyer information that comes there
